pybamm/models/submodels/interface/sei/ec_reaction_limited.py

```diff
@@ -0,0 +1,149 @@
+            self.domain +
+            " electrode EC surface concentration": c_ec,
+            self.domain +
+            " electrode EC surface concentration [mol.m-3]": c_ec * c_scale,
+            "X-averaged " + self.domain.lower() +
+            " electrode EC surface concentration": c_ec_av,
+            "X-averaged " + self.domain.lower() +
+            " electrode EC surface concentration": c_ec_av * c_scale,
+        }
+
+        return variables
+
+    def get_coupled_variables(self, variables):
+
+        # Update whole cell variables, which also updates the "sum of" variables
+        if (
+            "Negative electrode sei interfacial current density" in variables
+            and "Positive electrode sei interfacial current density" in variables
+            and "Sei interfacial current density" not in variables
+        ):
+            variables.update(
+                self._get_standard_whole_cell_interfacial_current_variables(variables)
+            )
+
+        return variables
+
+    def set_rhs(self, variables):
+        domain = self.domain.lower() + " electrode"
+        L_sei = variables["Total " + domain + " sei thickness"]
+        j_sei = variables[self.domain +
+                          " electrode sei interfacial current density"]
+
+        C_sei_j = pybamm.sei_parameters.C_sei_j
+
+        self.rhs = {L_sei: -j_sei * C_sei_j}
+
+    def set_algebraic(self, variables):
+        phi_s_n = variables[self.domain + " electrode potential"]
+        phi_e_n = variables[self.domain + " electrolyte potential"]
+        c_ec = variables[self.domain + " electrode EC surface concentration"]
+        j_sei = variables[self.domain +
+                          " electrode sei interfacial current density"]
+        L_sei = variables["Total " + self.domain.lower() +
+                          " electrode sei thickness"]
+
+        # Look for current that contributes to the -IR drop
+        # If we can't find the interfacial current density from the main reaction, j,
+        # it's ok to fall back on the total interfacial current density, j_tot
+        # This should only happen when the interface submodel is "InverseButlerVolmer"
+        # in which case j = j_tot (uniform) anyway
+        try:
+            j = variables["Total " +
+                          self.domain.lower() +
+                          " electrode interfacial current density"]
+        except KeyError:
+            j = variables["X-averaged " + self.domain.lower()
+                          + " electrode total interfacial current density"]
+        C_sei_ec = pybamm.sei_parameters.C_sei_ec
+        C_ec = pybamm.sei_parameters.C_ec
+        R_sei = pybamm.sei_parameters.R_sei
+
+        # need to revise for thermal case
+
+        self.algebraic = {j_sei: j_sei + C_sei_ec * c_ec * pybamm.exp(- 0.5 * (
+            phi_s_n - phi_e_n - j * L_sei * R_sei)),
+            c_ec: c_ec - pybamm.Scalar(1) - j_sei * L_sei * C_ec}
```

Comment:
it looks like `j_sei` and `c_ec` can be defined explicitly in terms of other variables in `get_coupled_variables`, instead of being made `Variable` objects themselves. While this will give the same answer, it's best to try to reduce the number of equations, especially algebraic

Reply:
c_ec and j_sei are present in both equations, and j_sei depends on itself because it is included in j (the total current). I think that makes it necessary to implement them with algebraic equations. It is however possible to replace c_ec in the equation of j_sei to just make one larger equation, but I think that makes it difficult to understand if the equations in Yang et. al. where correctly implemented. I would  do the latter after it is clear that all scales are multiplied correctly, and that a reference experiment can be reproduced with the added models.